apiserver/common/modelstatus.go

```diff
@@ -99,6 +99,11 @@ func (c *ModelStatusAPI) modelStatus(tag string) (params.ModelStatus, error) {
 		return status, errors.Trace(err)
 	}
 
+	modelApplications, err := ModelApplicationInfo(st)
```

Comment:
ModelApplicationInfo calls `st.AllApplications()`, which is also called on line 88 above. This should be unnecessary.

What is the difference between modelApplications and Applications? Only that modelApplications is transformed into a concrete struct. Applications should be dropped, and line 113 should read `ApplicationCount: len(modelApplications),`

Reply:
The same thing is going on with machines. `st.AllMachines()` is called twice unnecessarily